File: app/src/main/java/mariavv/fitnesspal/presentation/journal/JournalFragment.java
package mariavv.fitnesspal.presentation.journal;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.FragmentActivity;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.arellomobile.mvp.presenter.InjectPresenter;

import mariavv.fitnesspal.FitnessPal;
import mariavv.fitnesspal.R;
import mariavv.fitnesspal.other.Const;
import mariavv.fitnesspal.other.FrmFabric;

public class JournalFragment extends MvpAppCompatFragment implements JournalView, FrmFabric.IFragment {

    TextView dateTv;
    ImageView prevDayIv;
    ImageView nextDayIv;
    FloatingActionButton fab;

    ViewPager viewPager;

    PagerAdapter pagerAdapter;

    @InjectPresenter
    JournalPresenter presenter;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.fragment_journal, container, false);

        configureViews(view);

        return view;
    }

    private void configureViews(View view) {
        initViews(view);

        viewPager.setAdapter(pagerAdapter);

        prevDayIv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.onPrevDayClick(getCurrentPage(), getPageCount());
            }
        });
        nextDayIv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.onNextDayClick(getCurrentPage(), getPageCount());
            }
        });

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.onFabClick();
            }
        });
    }

    private void initViews(View view) {
        pagerAdapter = new PagerAdapter(getChildFragmentManager());
        viewPager = view.findViewById(R.id.viewPager);

        dateTv = view.findViewById(R.id.date);
        prevDayIv = view.findViewById(R.id.left);
        nextDayIv = view.findViewById(R.id.right);

        fab = view.findViewById(R.id.fab);
    }

    private int getPageCount() {
        return pagerAdapter.getCount();
    }

    private int getCurrentPage() {
        return viewPager.getCurrentItem();
    }

    @Override
    public void moveToPosition(int position, String date) {
        viewPager.setCurrentItem(position);
        dateTv.setText(date);
    }

    @Override
    public void setAdapterItems(long[] journalDates) {
        pagerAdapter.setItems(journalDates);
    }

    @Override
    public void setPrevDayEnable() {
        prevDayIv.setImageDrawable(FitnessPal.appContext.getDrawable(R.drawable.ic_chevron_left_black_24dp));
    }

    @Override
    public void setNextDayEnable() {
        nextDayIv.setImageDrawable(FitnessPal.appContext.getDrawable(R.drawable.ic_chevron_right_black_24dp));
    }

    @Override
    public void setPrevDayDisable() {
        prevDayIv.setImageDrawable(FitnessPal.appContext.getDrawable(R.drawable.ic_chevron_left_black_inactive_24dp));
    }

    @Override
    public void setNextDayDisable() {
        nextDayIv.setImageDrawable(FitnessPal.appContext.getDrawable(R.drawable.ic_chevron_right_black_inactive_24dp));
    }

    @Override
    public void setTitle() {
        final FragmentActivity activity = getActivity();
        if (activity != null) {
            activity.setTitle(R.string.journal_title);
        }
    }

    @Override
    public void setDate(String date) {
        dateTv.setText(date);
    }

    @Override
    public void onBackPressed() {
        presenter.onBackPressed();
    }

    @Override
    public String getName() {
        return Const.Screen.JOURNAL;
    }
}

File: app/src/main/java/mariavv/fitnesspal/presentation/journal/JournalFragment.kt
package mariavv.fitnesspal.presentation.journal

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.arellomobile.mvp.presenter.InjectPresenter
import kotlinx.android.synthetic.main.fragment_journal.*
import mariavv.fitnesspal.App
import mariavv.fitnesspal.R
import mariavv.fitnesspal.other.Const
import mariavv.fitnesspal.other.FrmFabric
import mariavv.fitnesspal.presentation.common.BaseFragment

class JournalFragment : BaseFragment(), JournalView, FrmFabric.IFragment {

    private lateinit var pagerAdapter: PagerAdapter

    @InjectPresenter
    lateinit var presenter: JournalPresenter

    private val pageCount: Int
        get() = pagerAdapter.count

    private val currentPage: Int
        get() = viewPager.currentItem

    override val name: String
        get() = Const.Screen.JOURNAL

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_journal, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        configureViews()
    }

    private fun configureViews() {
        pagerAdapter = PagerAdapter(childFragmentManager)
        viewPager.adapter = pagerAdapter

        left.setOnClickListener { presenter.onPrevDayClick(currentPage, pageCount) }
        right.setOnClickListener { presenter.onNextDayClick(currentPage, pageCount) }

        fab.setOnClickListener { presenter.onFabClick() }
    }

    override fun moveToPosition(position: Int, date: String) {
        viewPager.currentItem = position
        date_tv.text = date
    }

    override fun setAdapterItems(journalDates: LongArray) {
        pagerAdapter.setItems(journalDates)
    }

    override fun setPrevDayEnable() {
        left.setImageDrawable(App.context.getDrawable(R.drawable.ic_chevron_left_black_24dp))
    }

    override fun setNextDayEnable() {
        right.setImageDrawable(App.context.getDrawable(R.drawable.ic_chevron_right_black_24dp))
    }

    override fun setPrevDayDisable() {
        left.setImageDrawable(App.context.getDrawable(R.drawable.ic_chevron_left_black_inactive_24dp))
    }

    override fun setNextDayDisable() {
        right.setImageDrawable(App.context.getDrawable(R.drawable.ic_chevron_right_black_inactive_24dp))
    }

    override fun setDate(date: String) {
        date_tv.text = date
    }

    override fun onBackPressed() {
        presenter.onBackPressed()
    }
}
